homeassistant/components/vconnex/config_flow.py

```diff
@@ -0,0 +1,142 @@
+"""Config flow for Vconnex integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from vconnex.api import VconnexAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_PROJECT_NAME,
+    CONF_USER_ID,
+    DEFAULT_ENDPOINT,
+    DOMAIN,
+    DOMAIN_NAME,
+    PROJECT_CODE,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+TOKEN_USER_ID = "userId"
+TOKEN_PROJECT_NAME = "projectName"
+
+
+def validate_input(
+    hass: HomeAssistant, user_input: dict[str, Any]
+) -> dict[str, Any] | None:
+    """Validate the user input allows us to connect."""
+    client_id = user_input.get(CONF_CLIENT_ID)
+    client_secret = user_input.get(CONF_CLIENT_SECRET)
+
+    if None in (client_id, client_secret):
+        return None
+
+    if (
+        DOMAIN in hass.data
+        and len(datas := hass.data[DOMAIN].values())
+        and (client_id in map(lambda data: data.config_data.get(CONF_CLIENT_ID), datas))
+    ):
+        raise CredentialsUsed
+
+    try:
+        api = VconnexAPI(
+            DEFAULT_ENDPOINT, client_id, client_secret, project_code=PROJECT_CODE
+        )
+        if not api.is_valid():
+            LOGGER.error("Can't validate user credentials: %s", client_id)
+            raise InvalidCredentials
+    except InvalidCredentials as invalid_credential_exception:
+        raise invalid_credential_exception
```

Comment:
This just reraises the same error. It can be removed.

Reply:
Ah sorry, now I see it would be catched by the broad `except Exception` then. 
If it can't raise move it out of the try block.

Reply:
The `api.is_valid()` may raise connection error exception, which I dont know exactly. So I used broad-except to catch it